# 2. Teil - Praxis

Lese den Datensatz Crowdfunding ein.

Führe das Data PreProcessing und eine explorative Datenanalyse durch. (ggf. neue Features)

Welche Erkenntnisse können aus dem Datensatz gewonnen werden?

Verwende dabei verschiedene Plots. Gehe hierbei auch auf die Erkenntnisse ein, die aus dem Plot gewonnen werden



Der gesplittete Datensatz enthält folgende Spalten (inkl. Bedeutung):

    - funded_amount ... mit Ablauf der Crowdfundingzeit erhaltener Betrag/ ausgezahlter Betrag in USD
    - loan_amount ... Zielbetrag (Betrag dem man mit Funding erreichen wollte) in USD
    - activity ... Unterkategory zu dem das Ziel des Crowdfunding thematisch gehört
    - sector ... Oberkategory in den das Crowdfunding Thema fällt
    - use ... Kurzbeschreibung wofür das Geld verwendet werden soll
    - country_code ... Ländercode nach ISO Norm
    - country ... Ländername nach ISO Norm
    - region ... Region
    - currency ... Währung in den der funded_amount dann ausgezahlt wurde
    - partner_id ...zuordenbare ID zu einer Partnerorganisation
    - term in months ... Dauer über die der Kredit ausgezahlt wurde
    - lender_count ...Darlehensgeber (also wieviele Personen Geld für das Projekt gegeben haben
    - borrower_genders ... Geschlecht des Darlehensnehmer, also der das Crowdfunding initiiert hat
    - repayment interval ... Rückzahlungsmodalitäten/-rhythmus


## Anmerkung:
Quer-Checks, die überprüfen sollen, ob bestimmte Operationen richtig ausgeführt wurden, wurden der Übersichtlichkeit halber mit einem Hashtag auskommentiert. Zum Ansehen bitte Hashtag entfernen.

In [1]:
# Bibliothek importieren

import pandas as pd

# 1. Daten importieren
Nach erster Durchsicht der Datensätze fällt auf:
Teil1 ist durch # getrennt, Teil2 durch Kommmas. Die Spaltenbezeichnungen scheinen identisch zu sein. Datensatz Teil2 wird dem Datensatz Teil1 reihenweise (nach unten hin) angehängt. 
Anhand von Spalte 1 (Index - ohne Bezeichnung) lässt sich vermuten, dass der gesamte Datensatz 671.205 Zeilen (Index beginnt mit 0) besitzen dürfte (sofern keine Zeilen fehlen).

In [2]:
# crowdfunding_teil1.csv importieren

df_1 = pd.read_csv('crowdfunding_teil1.csv', sep='#')
df_1

Unnamed: 0  funded_amount  loan_amount               activity  \
0                0          300.0        300.0    Fruits & Vegetables   
1                1          575.0        575.0               Rickshaw   
2                2          150.0        150.0         Transportation   
3                3          200.0        200.0             Embroidery   
4                4          400.0        400.0             Milk Sales   
...            ...            ...          ...                    ...   
334995      334995         7125.0       7125.0         Plastics Sales   
334996      334996          125.0        125.0        Home Appliances   
334997      334997          925.0        925.0                 Cattle   
334998      334998          175.0        175.0          General Store   
334999      334999          150.0        150.0  Food Production/Sales   

                sector                                                use  \
0                 Food            To buy seasonal, fresh fruits to sell.    
1       Transportation  to repair and maintain the auto rickshaw used ...   
2       Transportation  To repair their old cycle-van and buy another ...   
3                 Arts  to purchase an embroidery machine and a variet...   
4                 Food                           to purchase one buffalo.   
...                ...                                                ...   
334995          Retail  to stock up with a dozen plastic tubs and buck...   
334996    Personal Use  to buy a water filter to provide safe drinking...   
334997     Agriculture  to increase his herd to get more milk and prov...   
334998          Retail  to buy additional items to sell, like beverage...   
334999            Food                      to buy sticky rice and sugar.   

       country_code                               country  \
0                PK                              Pakistan   
1                PK                              Pakistan   
2                IN                                 India   
3                PK                              Pakistan   
4                PK                              Pakistan   
...             ...                                   ...   
334995           CD  The Democratic Republic of the Congo   
334996           KH                              Cambodia   
334997           TJ                            Tajikistan   
334998           PH                           Philippines   
334999           PH                           Philippines   

                           region currency  partner_id  term_in_months  \
0                          Lahore      PKR       247.0            12.0   
1                          Lahore      PKR       247.0            11.0   
2                       Maynaguri      INR       334.0            43.0   
3                          Lahore      PKR       247.0            11.0   
4                    Abdul Hakeem      PKR       245.0            14.0   
...                           ...      ...         ...             ...   
334995  Goma, North Kivu province      USD       160.0             6.0   
334996                 Phnom Penh      KHR       311.0             8.0   
334997                   Khuroson      TJS        63.0            14.0   
334998                Palo, Leyte      PHP       145.0             8.0   
334999               Jagna, Bohol      PHP       125.0             8.0   

        lender_count                                   borrower_genders  \
0                 12                                             female   
1                 14                                     female, female   
2                  6                                             female   
3                  8                                             female   
4                 16                                             female   
...              ...                                                ...   
334995            15  female, female, female, female, femal

In [3]:
# zweiten Teil importieren:

df_2 = pd.read_csv('crowdfunding_teil2.csv', sep=',')
df_2

Unnamed: 0  funded_amount  loan_amount                    activity  \
0           335000          175.0        175.0  Liquor Store / Off-License   
1           335001          325.0        325.0                   Livestock   
2           335002          550.0        550.0                  Food Stall   
3           335003         1300.0       1300.0                      Cattle   
4           335004          900.0        900.0              Consumer Goods   
...            ...            ...          ...                         ...   
336200      671200            0.0         25.0                   Livestock   
336201      671201           25.0         25.0                   Livestock   
336202      671202            0.0         25.0                       Games   
336203      671203            0.0         25.0                   Livestock   
336204      671204            0.0         25.0                   Livestock   

               sector                                                use  \
0                Food  to purchase additional stock of coconut wine t...   
1         Agriculture         to buy 3 zebus and food to fatten them up.   
2                Food  to buy ingredients for her food-vending busine...   
3         Agriculture                         to buy one head of cattle.   
4        Personal Use              to buy consumer goods amongst others.   
...               ...                                                ...   
336200    Agriculture  [True, u'para compara: cemento, arenya y ladri...   
336201    Agriculture  [True, u'to start a turducken farm.'] - this l...   
336202  Entertainment                                                NaN   
336203    Agriculture  [True, u'to start a turducken farm.'] - this l...   
336204    Agriculture  [True, u'to start a turducken farm.'] - this l...   

       country_code      country                       region currency  \
0                PH  Philippines                  Palo, Leyte      PHP   
1                MG   Madagascar                    Antsirabe      MGA   
2                PH  Philippines                Cordova, Cebu      PHP   
3                EG        Egypt                     Baniswef      EGP   
4                PE         Peru  Urubamba - Urubamba - Cusco      PEN   
...             ...          ...                          ...      ...   
336200           PY     Paraguay                   Concepción      USD   
336201           KE        Kenya                          NaN      KES   
336202           KE        Kenya                          NaN      KES   
336203           KE        Kenya                          NaN      KES   
336204           KE        Kenya                          NaN      KES   

        partner_id  term_in_months  lender_count borrower_genders  \
0            145.0             8.0             6           female   
1            359.0            12.0            13           female   
2            145.0             5.0             6           female   
3            440.0            14.0            50             male   
4            119.0             6.0             1           female   
...            ...             ...           ...              ...   
336200        58.0            13.0             0           female   
336201       138.0            13.0             1           female   
336202       138.0            13.0             0              NaN   
336203       138.0            13.0             0           female   
336204       138.0            13.0             0           female   

       repayment_interval  
0               irregular  
1                 monthly  
2               irregular  
3                 monthly  
4               irregular  
...                   ...  
336200            monthly  
336201            monthly  
336202            monthly  
336203            monthly  
336204            monthly  

[336205 rows x 15 columns]

In [4]:
# Datensätze zusammenfügen

df = df_1.append(other=df_2)
df

Unnamed: 0  funded_amount  loan_amount             activity  \
0                0          300.0        300.0  Fruits & Vegetables   
1                1          575.0        575.0             Rickshaw   
2                2          150.0        150.0       Transportation   
3                3          200.0        200.0           Embroidery   
4                4          400.0        400.0           Milk Sales   
...            ...            ...          ...                  ...   
336200      671200            0.0         25.0            Livestock   
336201      671201           25.0         25.0            Livestock   
336202      671202            0.0         25.0                Games   
336203      671203            0.0         25.0            Livestock   
336204      671204            0.0         25.0            Livestock   

                sector                                                use  \
0                 Food            To buy seasonal, fresh fruits to sell.    
1       Transportation  to repair and maintain the auto rickshaw used ...   
2       Transportation  To repair their old cycle-van and buy another ...   
3                 Arts  to purchase an embroidery machine and a variet...   
4                 Food                           to purchase one buffalo.   
...                ...                                                ...   
336200     Agriculture  [True, u'para compara: cemento, arenya y ladri...   
336201     Agriculture  [True, u'to start a turducken farm.'] - this l...   
336202   Entertainment                                                NaN   
336203     Agriculture  [True, u'to start a turducken farm.'] - this l...   
336204     Agriculture  [True, u'to start a turducken farm.'] - this l...   

       country_code   country        region currency  partner_id  \
0                PK  Pakistan        Lahore      PKR       247.0   
1                PK  Pakistan        Lahore      PKR       247.0   
2                IN     India     Maynaguri      INR       334.0   
3                PK  Pakistan        Lahore      PKR       247.0   
4                PK  Pakistan  Abdul Hakeem      PKR       245.0   
...             ...       ...           ...      ...         ...   
336200           PY  Paraguay    Concepción      USD        58.0   
336201           KE     Kenya           NaN      KES       138.0   
336202           KE     Kenya           NaN      KES       138.0   
336203           KE     Kenya           NaN      KES       138.0   
336204           KE     Kenya           NaN      KES       138.0   

        term_in_months  lender_count borrower_genders repayment_interval  
0                 12.0            12           female          irregular  
1                 11.0            14   female, female          irregular  
2                 43.0             6           female             bullet  
3                 11.0             8           female          irregular  
4                 14.0            16           female            monthly  
...                ...           ...              ...                ...  
336200            13.0             0           female            monthly  
336201            13.0             1           female            monthly  
336202            13.0             0              NaN            monthly  
336203            13.0             0           female            monthly  
336204            13.0             0           female            monthly  

[671205 rows x 15 columns]

In [5]:
# prüfen, ob Datensätze richtig zusammengefügt wurden. Dafür lassen wir uns den Bereich ausgeben, an dem angefügt wurde

df.iloc[334990:335002, :]

Unnamed: 0  funded_amount  loan_amount                    activity  \
334990      334990         3675.0       3675.0                        Taxi   
334991      334991          800.0        800.0                  Food Stall   
334992      334992          900.0        900.0               General Store   
334993      334993         4350.0       4350.0                 Agriculture   
334994      334994          300.0        300.0                 Auto Repair   
334995      334995         7125.0       7125.0              Plastics Sales   
334996      334996          125.0        125.0             Home Appliances   
334997      334997          925.0        925.0                      Cattle   
334998      334998          175.0        175.0               General Store   
334999      334999          150.0        150.0       Food Production/Sales   
0           335000          175.0        175.0  Liquor Store / Off-License   
1           335001          325.0        325.0                   Livestock   

                sector                                                use  \
334990  Transportation                     to purchase tires and mirrors.   
334991            Food                  to buy corn for making tortillas.   
334992          Retail  to buy Rinso products (laundry detergent), sug...   
334993     Agriculture  to start producing and selling honey so he can...   
334994        Services  to purchase tools and spare parts for her husb...   
334995          Retail  to stock up with a dozen plastic tubs and buck...   
334996    Personal Use  to buy a water filter to provide safe drinking...   
334997     Agriculture  to increase his herd to get more milk and prov...   
334998          Retail  to buy additional items to sell, like beverage...   
334999            Food                      to buy sticky rice and sugar.   
0                 Food  to purchase additional stock of coconut wine t...   
1          Agriculture         to buy 3 zebus and food to fatten them up.   

       country_code                               country  \
334990           PE                                  Peru   
334991           GT                             Guatemala   
334992           SV                           El Salvador   
334993           BO                               Bolivia   
334994           PK                              Pakistan   
334995           CD  The Democratic Republic of the Congo   
334996           KH                              Cambodia   
334997           TJ                            Tajikistan   
334998           PH                           Philippines   
334999           PH                           Philippines   
0                PH                           Philippines   
1                MG                            Madagascar   

                           region currency  partner_id  term_in_months  \
334990                      Cusco      PEN       119.0             6.0   
334991                Villa Nueva      GTQ        97.0            10.0   
334992              Sensuntepeque      USD       199.0            17.0   
334993                 Monteagudo      BOB       140.0            63.0   
334994                 Rawalpindi      PKR       247.0            14.0   
334995  Goma, North Kivu province      USD       160.0             6.0   
334996                 Phnom Penh      KHR       311.0             8.0   
334997                   Khuroson      TJS        63.0            14.0   
334998                Palo, Leyte      PHP       145.0             8.0   
334999               Jagna, Bohol      PHP       125.0             8.0   
0                     Palo, Leyte      PHP       145.0             8.0   
1                       Antsirabe      MGA       359.0            12.0   

        lender_count                                   borrower_genders  \
334990             8  female, female, female, female, female, female...   
334991            22                                     female, female   
334992   

##### Beobachtung:
Man sieht, dass durch das Zusammenfügen ein Index-Sprung von 334999 auf 0 entstanden ist. Deshalb fügen wir eine neue Index-Spalte ein, da wir dann im Vergleich zur Spalte "Unnamed: 0" auch sehen können, ob ggf. Zeilen fehlen.

In [6]:
# neuen Index hinzufügen:

df = df.reset_index()
df

index  Unnamed: 0  funded_amount  loan_amount             activity  \
0            0           0          300.0        300.0  Fruits & Vegetables   
1            1           1          575.0        575.0             Rickshaw   
2            2           2          150.0        150.0       Transportation   
3            3           3          200.0        200.0           Embroidery   
4            4           4          400.0        400.0           Milk Sales   
...        ...         ...            ...          ...                  ...   
671200  336200      671200            0.0         25.0            Livestock   
671201  336201      671201           25.0         25.0            Livestock   
671202  336202      671202            0.0         25.0                Games   
671203  336203      671203            0.0         25.0            Livestock   
671204  336204      671204            0.0         25.0            Livestock   

                sector                                                use  \
0                 Food            To buy seasonal, fresh fruits to sell.    
1       Transportation  to repair and maintain the auto rickshaw used ...   
2       Transportation  To repair their old cycle-van and buy another ...   
3                 Arts  to purchase an embroidery machine and a variet...   
4                 Food                           to purchase one buffalo.   
...                ...                                                ...   
671200     Agriculture  [True, u'para compara: cemento, arenya y ladri...   
671201     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671202   Entertainment                                                NaN   
671203     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671204     Agriculture  [True, u'to start a turducken farm.'] - this l...   

       country_code   country        region currency  partner_id  \
0                PK  Pakistan        Lahore      PKR       247.0   
1                PK  Pakistan        Lahore      PKR       247.0   
2                IN     India     Maynaguri      INR       334.0   
3                PK  Pakistan        Lahore      PKR       247.0   
4                PK  Pakistan  Abdul Hakeem      PKR       245.0   
...             ...       ...           ...      ...         ...   
671200           PY  Paraguay    Concepción      USD        58.0   
671201           KE     Kenya           NaN      KES       138.0   
671202           KE     Kenya           NaN      KES       138.0   
671203           KE     Kenya           NaN      KES       138.0   
671204           KE     Kenya           NaN      KES       138.0   

        term_in_months  lender_count borrower_genders repayment_interval  
0                 12.0            12           female          irregular  
1                 11.0            14   female, female          irregular  
2                 43.0             6           female             bullet  
3                 11.0             8           female          irregular  
4                 14.0            16           female            monthly  
...                ...           ...              ...                ...  
671200            13.0             0           female            monthly  
671201            13.0             1           female            monthly  
671202            13.0             0              NaN            monthly  
671203            13.0             0           female            monthly  
671204            13.0             0           female            monthly  

[671205 rows x 16 columns]

In [7]:
# nochmal überpüfen: 

# df.iloc[334990:335002, :]

In [8]:
# alter Index (Spalte 'index') wird gelöscht:

df.drop(columns='index', inplace=True)

In [9]:
df

Unnamed: 0  funded_amount  loan_amount             activity  \
0                0          300.0        300.0  Fruits & Vegetables   
1                1          575.0        575.0             Rickshaw   
2                2          150.0        150.0       Transportation   
3                3          200.0        200.0           Embroidery   
4                4          400.0        400.0           Milk Sales   
...            ...            ...          ...                  ...   
671200      671200            0.0         25.0            Livestock   
671201      671201           25.0         25.0            Livestock   
671202      671202            0.0         25.0                Games   
671203      671203            0.0         25.0            Livestock   
671204      671204            0.0         25.0            Livestock   

                sector                                                use  \
0                 Food            To buy seasonal, fresh fruits to sell.    
1       Transportation  to repair and maintain the auto rickshaw used ...   
2       Transportation  To repair their old cycle-van and buy another ...   
3                 Arts  to purchase an embroidery machine and a variet...   
4                 Food                           to purchase one buffalo.   
...                ...                                                ...   
671200     Agriculture  [True, u'para compara: cemento, arenya y ladri...   
671201     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671202   Entertainment                                                NaN   
671203     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671204     Agriculture  [True, u'to start a turducken farm.'] - this l...   

       country_code   country        region currency  partner_id  \
0                PK  Pakistan        Lahore      PKR       247.0   
1                PK  Pakistan        Lahore      PKR       247.0   
2                IN     India     Maynaguri      INR       334.0   
3                PK  Pakistan        Lahore      PKR       247.0   
4                PK  Pakistan  Abdul Hakeem      PKR       245.0   
...             ...       ...           ...      ...         ...   
671200           PY  Paraguay    Concepción      USD        58.0   
671201           KE     Kenya           NaN      KES       138.0   
671202           KE     Kenya           NaN      KES       138.0   
671203           KE     Kenya           NaN      KES       138.0   
671204           KE     Kenya           NaN      KES       138.0   

        term_in_months  lender_count borrower_genders repayment_interval  
0                 12.0            12           female          irregular  
1                 11.0            14   female, female          irregular  
2                 43.0             6           female             bullet  
3                 11.0             8           female          irregular  
4                 14.0            16           female            monthly  
...                ...           ...              ...                ...  
671200            13.0             0           female            monthly  
671201            13.0             1           female            monthly  
671202            13.0             0              NaN            monthly  
671203            13.0             0           female            monthly  
671204            13.0             0           female            monthly  

[671205 rows x 15 columns]

In [10]:
# Ursprüngliche Index-Spalte aus den CSVs umbenennen in 'id'
# Spalte wird beibehalten, um vorübergehend einen Abgleich zu haben, falls mitten im Datensatz ein Dreher sein sollten.
# Das ermöglicht ggf. später einen Abgleich durchzuführen.

# Umbenennung der Spalte loan_amount in 'funding_goal' für leichtere Verständlichkeit

df.rename(columns={
                    'Unnamed: 0': 'id',
                    'loan_amount': 'funding_goal'
                  }, inplace=True)
df

id  funded_amount  funding_goal             activity  \
0            0          300.0         300.0  Fruits & Vegetables   
1            1          575.0         575.0             Rickshaw   
2            2          150.0         150.0       Transportation   
3            3          200.0         200.0           Embroidery   
4            4          400.0         400.0           Milk Sales   
...        ...            ...           ...                  ...   
671200  671200            0.0          25.0            Livestock   
671201  671201           25.0          25.0            Livestock   
671202  671202            0.0          25.0                Games   
671203  671203            0.0          25.0            Livestock   
671204  671204            0.0          25.0            Livestock   

                sector                                                use  \
0                 Food            To buy seasonal, fresh fruits to sell.    
1       Transportation  to repair and maintain the auto rickshaw used ...   
2       Transportation  To repair their old cycle-van and buy another ...   
3                 Arts  to purchase an embroidery machine and a variet...   
4                 Food                           to purchase one buffalo.   
...                ...                                                ...   
671200     Agriculture  [True, u'para compara: cemento, arenya y ladri...   
671201     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671202   Entertainment                                                NaN   
671203     Agriculture  [True, u'to start a turducken farm.'] - this l...   
671204     Agriculture  [True, u'to start a turducken farm.'] - this l...   

       country_code   country        region currency  partner_id  \
0                PK  Pakistan        Lahore      PKR       247.0   
1                PK  Pakistan        Lahore      PKR       247.0   
2                IN     India     Maynaguri      INR       334.0   
3                PK  Pakistan        Lahore      PKR       247.0   
4                PK  Pakistan  Abdul Hakeem      PKR       245.0   
...             ...       ...           ...      ...         ...   
671200           PY  Paraguay    Concepción      USD        58.0   
671201           KE     Kenya           NaN      KES       138.0   
671202           KE     Kenya           NaN      KES       138.0   
671203           KE     Kenya           NaN      KES       138.0   
671204           KE     Kenya           NaN      KES       138.0   

        term_in_months  lender_count borrower_genders repayment_interval  
0                 12.0            12           female          irregular  
1                 11.0            14   female, female          irregular  
2                 43.0             6           female             bullet  
3                 11.0             8           female          irregular  
4                 14.0            16           female            monthly  
...                ...           ...              ...                ...  
671200            13.0             0           female            monthly  
671201            13.0             1           female            monthly  
671202            13.0             0              NaN            monthly  
671203            13.0             0           female            monthly  
671204            13.0             0           female            monthly  

[671205 rows x 15 columns]

# 2. Data PreProcessing
### 2.1 Datensatz genauer ansehen

##### Missings:

Use: 
-	NaN
-	Eckige Klammern

##### Fehler im Datensatz:
Borrower_genders:
- Es gibt Dopplungen vom Geschlecht: female, female, female  redudanzen entfernen
- bei currency ist 'monthly' drin ???

##### Beobachtungen: 
- Wenn lender_count = 0, müsste funded_amount auch = 0 sein: wenn es niemanden gibt, der Geld gibt, dann kann es auch keinen Funding-Betrag geben.
- country_code scheint der genormte ISO-Country-Code zu sein.

##### neue Spalten:
- count_initiators: wieviele Initiatoren das Projekt leiten (Anzahl der Einträge in der Spalte borrower_genders)
- gender_initiators: weiblich/männlich/gemischt (evtl.)
- successful: 
    yes = funded_amount >= loan_amount
    inbetween: 0 < funded_amount < loan_amount (sollte nicht existieren nach Crowdfunding-Logik)
    no = funded_amount = 0
- use_angegeben: ja/nein (non-null / null)

## 2.2 Missings

### 2.2.1 Übersicht
Zusammenfassung nachfolgender Ergebnisse am Ende des Abschnitts.

In [11]:
# numpy importieren, um mit NaN-Werten besser umzugehen (nicht undbedingt nötig)
import numpy as np

In [12]:
# Datentypen prüfen, Null-Werte und Anzahl der Einträge (Übersicht)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 15 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   funding_goal        671205 non-null  float64
 3   activity            671205 non-null  object 
 4   sector              671205 non-null  object 
 5   use                 666973 non-null  object 
 6   country_code        671197 non-null  object 
 7   country             671205 non-null  object 
 8   region              614405 non-null  object 
 9   currency            671205 non-null  object 
 10  partner_id          657698 non-null  float64
 11  term_in_months      671205 non-null  float64
 12  lender_count        671205 non-null  int64  
 13  borrower_genders    666984 non-null  object 
 14  repayment_interval  671205 non-null  object 
dtypes: float64(4), int64(2), object(9)

In [13]:
df.isnull().sum()

id                        0
funded_amount             0
funding_goal              0
activity                  0
sector                    0
use                    4232
country_code              8
country                   0
region                56800
currency                  0
partner_id            13507
term_in_months            0
lender_count              0
borrower_genders       4221
repayment_interval        0
dtype: int64

In [14]:
# Ausprägungen der Spalten anzeigen (unique-Werte), um zu sehen, wie Missings kodiert sind:

df.use.unique

# hat NaN

<bound method Series.unique of 0                   To buy seasonal, fresh fruits to sell. 
1         to repair and maintain the auto rickshaw used ...
2         To repair their old cycle-van and buy another ...
3         to purchase an embroidery machine and a variet...
4                                  to purchase one buffalo.
                                ...                        
671200    [True, u'para compara: cemento, arenya y ladri...
671201    [True, u'to start a turducken farm.'] - this l...
671202                                                  NaN
671203    [True, u'to start a turducken farm.'] - this l...
671204    [True, u'to start a turducken farm.'] - this l...
Name: use, Length: 671205, dtype: object>

In [15]:
df.country_code.unique()

# hat nan --> auffüllen anhand der Länder

array(['PK', 'IN', 'KE', 'NI', 'SV', 'TZ', 'PH', 'PE', 'SN', 'KH', 'LR',
       'VN', 'IQ', 'HN', 'PS', 'MN', 'US', 'ML', 'CO', 'TJ', 'GT', 'EC',
       'BO', 'YE', 'GH', 'SL', 'HT', 'CL', 'JO', 'UG', 'BI', 'BF', 'TL',
       'ID', 'GE', 'UA', 'XK', 'AL', 'CD', 'CR', 'SO', 'ZW', 'CM', 'TR',
       'AZ', 'DO', 'BR', 'MX', 'KG', 'AM', 'PY', 'LB', 'WS', 'IL', 'RW',
       'ZM', 'NP', 'CG', 'MZ', 'ZA', 'TG', 'BJ', 'BZ', 'SR', 'TH', 'NG',
       'MR', 'VU', 'PA', 'VI', 'VC', 'LA', 'MW', 'MM', 'MD', 'SS', 'SB',
       'CN', 'EG', 'GU', 'AF', 'MG', nan, 'PR', 'LS', 'CI', 'BT'],
      dtype=object)

In [16]:
df.region.unique

<bound method Series.unique of 0               Lahore
1               Lahore
2            Maynaguri
3               Lahore
4         Abdul Hakeem
              ...     
671200      Concepción
671201             NaN
671202             NaN
671203             NaN
671204             NaN
Name: region, Length: 671205, dtype: object>

In [17]:
df.partner_id.unique()

array([247., 334., 245.,  nan, 164.,  96.,  81.,  87., 126., 145., 139.,
       108.,   9., 182., 331., 123., 166., 201., 122., 241., 116.,  61.,
        93., 203., 249.,  98.,  78., 177.,  63., 244., 113., 204., 167.,
       154., 137., 110., 205., 140., 106., 100., 231., 138., 199., 190.,
        59., 183., 269., 173.,  85.,  44., 157., 310., 342.,  80., 185.,
       343., 162., 121., 159., 187., 144., 243., 219., 105., 119., 125.,
        71., 259., 181., 136., 225.,  26., 238., 240., 239., 143.,  82.,
       160., 151., 133., 325., 118., 176., 172., 215., 184., 217., 198.,
       222.,  56., 298., 337., 294., 257.,  57., 156., 120., 171., 309.,
       163., 304., 129., 169.,  40., 228.,  48., 330.,  58.,  55., 115.,
       328., 321.,  15., 175., 112., 161.,  74., 226., 207., 268., 329.,
       148., 127., 150.,  97., 323., 146., 246., 242., 277., 299.,  42.,
       326., 210., 336., 212., 322., 292., 130., 260., 258.,  30., 230.,
       316.,  62., 165., 131.,  77., 279., 301., 31

In [18]:
df.borrower_genders.unique

<bound method Series.unique of 0                 female
1         female, female
2                 female
3                 female
4                 female
               ...      
671200            female
671201            female
671202               NaN
671203            female
671204            female
Name: borrower_genders, Length: 671205, dtype: object>

#### Erkenntnisse:
- Datentypen scheinen korrekt zu sein

- Non-Null gibt Ausfkunft über Missings (s.u.)

##### 1. fehlende Werte in:

- use (4.232) --> NaN (fiel bereits beim Betrachten des df auf) --> nicht auffüllbar, da unsystematischer Fließtext, Zeilen müssen nicht gelöscht werden, da use nur rein informativen Gehalt hat


- country_code (8) --> nan --> auffüllen anhand von country (s. 2.2.2)


- region (56.800 - viele! ca. 9%) --> NaN --> ansehen, ob die sich die Missings der Regionen in bestimmten Ländern häufen (s. 2.2.3)


- partner_id (13.507) --> nan --> anzunehmen ist, dass nan bedeutet, dass für dieses Crowdfunding-Projekt tatsächlich kein Partner mit im Boot ist. --> nan-Werte ersetzen durch 'no partner'(s. 2.2.4), Bemerkung: partner_id war als float gekennzeichnet, allerdings wird man mit einer ID (Art Kundennummer) nicht rechnen, von daher wird aus partner_id mit dem Einfügen von 'no partner' ein object.


- borrower_genders (4.221) --> NaN --> mit der Spalte wird eh nicht gerechnet: neue Spalten angeben: (s. 2.2.5)
        - Anzahl der Initiatoren
        - Geschlechterverteilung der Initiatoren: female, male, mixed


###### 2. Vollständige Spalten (geprüft mit unique(), aber da nicht auffällig aus Code wieder entfernt):
- funded_amount
- funding_goal
- activity
- sector
- country
- currency
- term_in_months
- lender_count
- repayment_interval (praktisch: kategoriale Variable mit 4 Ausprägungen)

###### 3. Die schleierhafte Spalte partner_id:
- These: dahinter verbirgt sich eine Art Kundennummer einer Partnerorganisation, mit der die Initiatoren kooperiert.


In [19]:
# prüfen, ob es ganz klare Korrelationen gibt, anhand derer man auffüllen kann. 
df.corr()

id  funded_amount  funding_goal  partner_id  \
id              1.000000      -0.050624     -0.032047    0.154690   
funded_amount  -0.050624       1.000000      0.945044   -0.075276   
funding_goal   -0.032047       0.945044      1.000000   -0.071251   
partner_id      0.154690      -0.075276     -0.071251    1.000000   
term_in_months -0.041399       0.149310      0.184795    0.094878   
lender_count   -0.099205       0.849168      0.798697   -0.008575   

                term_in_months  lender_count  
id                   -0.041399     -0.099205  
funded_amount         0.149310      0.849168  
funding_goal          0.184795      0.798697  
partner_id            0.094878     -0.008575  
term_in_months        1.000000      0.227283  
lender_count          0.227283      1.000000

In [20]:
# Variable, mit der man interessante Berechnungen machen kann
df.repayment_interval.unique()

array(['irregular', 'bullet', 'monthly', 'weekly'], dtype=object)

### 2.2.2 Behandung der Missings in country_code

In [21]:
df[df.country_code.isnull()]

id  funded_amount  funding_goal                   activity  \
202537  202537         4150.0        4150.0                  Wholesale   
202823  202823         4150.0        4150.0                  Wholesale   
344929  344929         3325.0        3325.0                  Wholesale   
351177  351177         3325.0        3325.0                  Wholesale   
420953  420953         3325.0        3325.0                  Wholesale   
421218  421218         4000.0        4000.0                  Wholesale   
487207  487207         5100.0        5100.0  Renewable Energy Products   
487653  487653         5000.0        5000.0                  Wholesale   

           sector                                                use  \
202537  Wholesale  To purchase lighting products for sale to loca...   
202823  Wholesale  To purchase lighting products for sale to loca...   
344929  Wholesale  To purchase lighting products for sale to loca...   
351177  Wholesale  To purchase lighting products for sale to loca...   
420953  Wholesale  To purchase lighting products for sale to loca...   
421218  Wholesale  purchase solar lighting products for sale to l...   
487207     Retail  to pay for stock of solar lights and cell phon...   
487653  Wholesale  to maintain a stock of solar lights and cell p...   

       country_code  country         region currency  partner_id  \
202537          NaN  Namibia        EEnhana      NAD       437.0   
202823          NaN  Namibia          Rundu      NAD       437.0   
344929          NaN  Namibia        EEnhana      NAD       437.0   
351177          NaN  Namibia          Rundu      NAD       437.0   
420953          NaN  Namibia        EEnhana      NAD       437.0   
421218          NaN  Namibia          Rundu      NAD       437.0   
487207          NaN  Namibia  Katima Mulilo      NAD       437.0   
487653          NaN  Namibia       Oshakati      NAD       437.0   

        term_in_months  lender_count borrower_genders repayment_interval  
202537             6.0           162           female             bullet  
202823             6.0           159             male             bullet  
344929             7.0           120           female             bullet  
351177             7.0           126             male             bullet  
420953             7.0           118           female             bullet  
421218             7.0           150             male             bullet  
487207             7.0           183             male             bullet  
487653             7.0           183           female             bullet

In [22]:
# sind nur Namibia --> ISO country code für Namibia laut google: NA
# ersetze alle NaNs mit NAM

df.loc[df.country_code.isnull(), 'country_code'] = 'NA'

In [23]:
df.country_code.unique()

array(['PK', 'IN', 'KE', 'NI', 'SV', 'TZ', 'PH', 'PE', 'SN', 'KH', 'LR',
       'VN', 'IQ', 'HN', 'PS', 'MN', 'US', 'ML', 'CO', 'TJ', 'GT', 'EC',
       'BO', 'YE', 'GH', 'SL', 'HT', 'CL', 'JO', 'UG', 'BI', 'BF', 'TL',
       'ID', 'GE', 'UA', 'XK', 'AL', 'CD', 'CR', 'SO', 'ZW', 'CM', 'TR',
       'AZ', 'DO', 'BR', 'MX', 'KG', 'AM', 'PY', 'LB', 'WS', 'IL', 'RW',
       'ZM', 'NP', 'CG', 'MZ', 'ZA', 'TG', 'BJ', 'BZ', 'SR', 'TH', 'NG',
       'MR', 'VU', 'PA', 'VI', 'VC', 'LA', 'MW', 'MM', 'MD', 'SS', 'SB',
       'CN', 'EG', 'GU', 'AF', 'MG', 'NA', 'PR', 'LS', 'CI', 'BT'],
      dtype=object)

### 2.2.3 Behandlung der Missings in regions

In [24]:
df_missing_regions = df[df.region.isnull()]

In [25]:
df_missing_regions.country.unique()

array(['Kenya', 'El Salvador', 'Senegal', 'Iraq', 'United States', 'Peru',
       'Tanzania', 'Guatemala', 'Colombia', 'Indonesia', 'Kosovo',
       'Timor-Leste', 'Turkey', 'Philippines', 'Palestine', 'Burundi',
       'Tajikistan', 'Honduras', 'Jordan', 'Mexico', 'Lebanon', 'Albania',
       'Nicaragua', 'Bolivia', 'Israel', 'Rwanda', 'Azerbaijan',
       'Ecuador', 'Mongolia', 'Haiti', 'Cambodia', 'Sierra Leone',
       'Yemen', 'Zimbabwe', 'Paraguay', 'Uganda', 'Armenia',
       'Dominican Republic', 'Benin', 'Belize', 'Ghana', 'Mozambique',
       'Zambia', 'Samoa', 'Brazil', 'Panama', 'Pakistan', 'Burkina Faso',
       'Suriname', 'Virgin Islands', 'Togo', 'South Africa', 'Malawi',
       'Nigeria', 'Liberia', 'Vietnam', 'Costa Rica', 'Guam',
       'Myanmar (Burma)', 'Mali', 'Madagascar',
       'The Democratic Republic of the Congo', 'Cameroon', 'Georgia',
       'Puerto Rico', 'South Sudan', 'Moldova', 'Chile', 'Kyrgyzstan',
       'India', 'China', 'Bhutan'], dtype=object)

In [26]:
df_missing_regions.country.value_counts()

El Salvador      20251
Kenya             8752
Rwanda            6138
United States     5173
Senegal           3231
                 ...  
Georgia              1
Kyrgyzstan           1
China                1
Chile                1
Bhutan               1
Name: country, Length: 72, dtype: int64

In [27]:
# theoretisch könnte man step-by-step die Top-Regions für jedes Land eruieren und anhand der häufigsten Region auffüllen
# sofern eine Region deutlich überwiegt. Es ist natürlich auch möglich, dass manche Regionen ihre Region angeben und andere
# nicht und deshalb die missing values eben nicht den ansonsten häufigsten Regionen des Landes entsprechen.

# Da eher auf Länderebene weiter gearbeitet werden soll, wird aus zeitlichen Gründen davon abgesehen.

### 2.2.4 Behandlung der Missings in partner_id

In [28]:
# ersetze nan-Werte durch 'no partner'

df.loc[df.partner_id.isnull(), 'partner_id'] = 'no partner'

In [29]:
df.loc[df.partner_id == 'no partner', :]

id  funded_amount  funding_goal               activity    sector  \
5            5          250.0         250.0               Services  Services   
67          67          125.0         125.0                 Energy  Services   
99          99         5000.0        5000.0  Food Production/Sales      Food   
114        114          125.0         125.0                 Energy  Services   
195        195          125.0         125.0          Grocery Store      Food   
...        ...            ...           ...                    ...       ...   
670825  670825            0.0        2000.0      Cleaning Services  Services   
670833  670833          500.0       10000.0                 Retail    Retail   
670883  670883         1350.0        2000.0          Grocery Store      Food   
670914  670914         1000.0        7000.0                   Arts      Arts   
670915  670915          500.0        2000.0             Restaurant      Food   

                                                      use country_code  \
5       purchase leather for my business using ksh 20000.           KE   
67                    purchase solar lanterns for resale.           KE   
99      hire more staff to keep up with the volume of ...           US   
114                          buy solar lanterns for sale.           KE   
195                       ncrease my yams stock for sales           KE   
...                                                   ...          ...   
670825  me purchase a new commercial vacuum cleaner an...           US   
670833          me remodel and expand my new storefront.            US   
670883  me to to buy more fruit, prep materials and pu...           US   
670914  us get a U.S. patent, and update our website &...           US   
670915                     me to build out my food truck!           US   

              country region currency  partner_id  term_in_months  \
5               Kenya    NaN      KES  no partner             4.0   
67              Kenya    NaN      KES  no partner             3.0   
99      United States    NaN      USD  no partner            30.0   
114             Kenya    NaN      KES  no partner             3.0   
195             Kenya    NaN      KES  no partner             3.0   
...               ...    ...      ...         ...             ...   
670825  United States    NaN      USD  no partner            18.0   
670833  United States    NaN      USD  no partner            36.0   
670883  United States    NaN      USD  no partner            18.0   
670914  United States    NaN      USD  no partner            36.0   
670915  United States    NaN      USD  no partner            18.0   

        lender_count borrower_genders repayment_interval  
5                  6           female          irregular  
67                 6             male          irregular  
99               261           female            monthly  
114                7           female          irregular  
195                6           female          irregular  
...              ...              ...                ...  
670825             0           female             bullet  
670833             9           female             bullet  
670883            21             male             bullet  
670914             7           female             bullet  
670915            10           female             bullet  

[13507 rows x 15 columns]

In [30]:
df.info()

# partner_id wurde zu object geändert

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 15 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   funding_goal        671205 non-null  float64
 3   activity            671205 non-null  object 
 4   sector              671205 non-null  object 
 5   use                 666973 non-null  object 
 6   country_code        671205 non-null  object 
 7   country             671205 non-null  object 
 8   region              614405 non-null  object 
 9   currency            671205 non-null  object 
 10  partner_id          671205 non-null  object 
 11  term_in_months      671205 non-null  float64
 12  lender_count        671205 non-null  int64  
 13  borrower_genders    666984 non-null  object 
 14  repayment_interval  671205 non-null  object 
dtypes: float64(3), int64(2), object(10

## 2.2.5 Behandlung der Missings in borrower_genders
1. Da die Nan-Werte nur 0,6% des Gesamtdatensatzes ausmachen und mit dieser Spalte weiter gearbeitet werden soll, werden die Zeilen der fehlenden Werte gelöscht. Ein anderes (systematisches) Auffüllen der Nans ist hier nicht möglich, da die Daten keinerlei Rückschlüsse darauf zulassen, wieviele und welche Geschlechter die Projekte initiierten.

In [31]:
# Nans ausgeben lassen
df.loc[df.borrower_genders.isnull(), 'borrower_genders']

140       NaN
145       NaN
170       NaN
412       NaN
414       NaN
         ... 
671151    NaN
671174    NaN
671178    NaN
671185    NaN
671202    NaN
Name: borrower_genders, Length: 4221, dtype: object

In [32]:
# Ansprechen des Index der Nans
df.loc[df.borrower_genders.isnull()].index

Int64Index([   140,    145,    170,    412,    414,    464,    466,    546,
               595,    616,
            ...
            658799, 659387, 659604, 660788, 661718, 671151, 671174, 671178,
            671185, 671202],
           dtype='int64', length=4221)

In [33]:
# Nans anhand ihres Index droppen (gesamte Zeile)
df.drop(index=df.loc[df.borrower_genders.isnull()].index, inplace=True)

In [34]:
# Test 1: keine Nans mehr vorhanden
df.loc[df.borrower_genders.isnull(), 'borrower_genders']

Series([], Name: borrower_genders, dtype: object)

In [35]:
# Test 2
df.borrower_genders.isnull

<bound method Series.isnull of 0                 female
1         female, female
2                 female
3                 female
4                 female
               ...      
671199            female
671200            female
671201            female
671203            female
671204            female
Name: borrower_genders, Length: 666984, dtype: object>

### Datentypen ändern

In [36]:
df = df.astype({"id": int, "funded_amount": int, "funding_goal": int, "term_in_months": int, "lender_count": int})

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 666984 entries, 0 to 671204
Data columns (total 15 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   id                  666984 non-null  int32 
 1   funded_amount       666984 non-null  int32 
 2   funding_goal        666984 non-null  int32 
 3   activity            666984 non-null  object
 4   sector              666984 non-null  object
 5   use                 666973 non-null  object
 6   country_code        666984 non-null  object
 7   country             666984 non-null  object
 8   region              614405 non-null  object
 9   currency            666984 non-null  object
 10  partner_id          666984 non-null  object
 11  term_in_months      666984 non-null  int32 
 12  lender_count        666984 non-null  int32 
 13  borrower_genders    666984 non-null  object
 14  repayment_interval  666984 non-null  object
dtypes: int32(5), object(10)
memory usage: 68.7+ MB


# 3. Neue Spalten

- 3.1 successful_funding: sinnvoll, um auszuwerten, wann funding stattfindet und wann nicht
    - no_funding: wenn funded_amount = 0
    - goal_not_achieved: wenn größer 0 und kleiner funding_goal
    - goal_achieved: wenn funded_amount = funding_goal
    - goal_overachieved: wenn funded_amount > funding_goal
- 3.2 use_description: sinnvoll, um zu sehen, ob Projekte mit Beschreibung eher Funding erhalten als ohne Beschreibung
    - ja/nein (non-null / null)
- 3.3 count_initiators: wieviele Initiatoren das Projekt leiten (Anzahl der Einträge in der Spalte borrower_genders)
- 3.4 gender_initiators: weiblich/männlich/gemischt (evtl.)

## 3.1 Spalte "successful_funding"

Zuerst wird untersucht, ob es auch Fälle gibt, bei denen zwar der funded_amount > 0 ist, aber das funding_goal nicht erreicht wurde. --> Auswertung unten zeigt, dass dies der Fall ist.

In [38]:
df.insert(loc=3 , column='successful_funding', value='goal_not_achieved')

In [39]:
df.loc[df.funded_amount == 0.0, 'successful_funding'] = 'no_funding'
#df

In [40]:
df.loc[df.funded_amount == df.funding_goal, 'successful_funding'] = 'goal_achieved'
#df

In [41]:
df.loc[df.funded_amount > df.funding_goal, 'successful_funding'] = 'goal_overachieved'
#df

In [42]:
df.successful_funding.value_counts()

# relativ viele Kampagnen haben ihr Funding-Ziel erreicht!

goal_achieved        619336
goal_not_achieved     44392
no_funding             3254
goal_overachieved         2
Name: successful_funding, dtype: int64

## 3.2 Spalte "use_description"
0 = no use description existant

1 = description existant

In [43]:
df.insert(loc=7 , column='use_description', value=1)

In [44]:
df.loc[df.use.isnull(), 'use_description'] = 0

In [45]:
#df

## 3.3 & 3.4 Spalten "count_initiators" und "gender_initiators"

### Vorbereitung: Spalten "count_male" und "count_female"

In [46]:
# Location ermitteln, nach dem die Spalte eingefügt werden soll
df.columns.get_loc('borrower_genders')

15

In [47]:
# Liste erstellen aus der Spalte borrower_genders:
gender_list = df.borrower_genders.tolist()

In [48]:
# zwei Spalten einfügen in df: eine für Frauen, eine für Männer
df.insert(16,'count_male' ,0,True)
df.insert(17,'count_female',0, True)

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 666984 entries, 0 to 671204
Data columns (total 19 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   id                  666984 non-null  int32 
 1   funded_amount       666984 non-null  int32 
 2   funding_goal        666984 non-null  int32 
 3   successful_funding  666984 non-null  object
 4   activity            666984 non-null  object
 5   sector              666984 non-null  object
 6   use                 666973 non-null  object
 7   use_description     666984 non-null  int64 
 8   country_code        666984 non-null  object
 9   country             666984 non-null  object
 10  region              614405 non-null  object
 11  currency            666984 non-null  object
 12  partner_id          666984 non-null  object
 13  term_in_months      666984 non-null  int32 
 14  lender_count        666984 non-null  int32 
 15  borrower_genders    666984 non-null  object
 16  co

In [50]:
# Datentypen nochmal auf integer ändern bevor wir die aufwändige Schleife rechnen:

df = df.astype({"use_description": int, "count_male": int, "count_female": int})

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 666984 entries, 0 to 671204
Data columns (total 19 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   id                  666984 non-null  int32 
 1   funded_amount       666984 non-null  int32 
 2   funding_goal        666984 non-null  int32 
 3   successful_funding  666984 non-null  object
 4   activity            666984 non-null  object
 5   sector              666984 non-null  object
 6   use                 666973 non-null  object
 7   use_description     666984 non-null  int32 
 8   country_code        666984 non-null  object
 9   country             666984 non-null  object
 10  region              614405 non-null  object
 11  currency            666984 non-null  object
 12  partner_id          666984 non-null  object
 13  term_in_months      666984 non-null  int32 
 14  lender_count        666984 non-null  int32 
 15  borrower_genders    666984 non-null  object
 16  co

In [ ]:
# dieser Schritt wurde nicht erneut vor Export in HTML ausgeführt, da diese Berechnung mehrere Stunden benötigt. 
# Im nächsten Jupyter Notebook wird jedoch mit dem neuen Datensatz, welcher am Ende exportiert wurde, weiter gerechnet.
# leere temporäre Liste & Counter

temp_list = []
counter = 0

# for-Schleife
for i in gender_list:
    temp_list.clear()
    temp_list = str(i).split(', ')
    temp_male = temp_list.count('male') 
    temp_female = temp_list.count('female')
    df.iloc[counter,16] = temp_male
    df.iloc[counter,17] = temp_female
    counter = counter+1

### Fortsetzung in Notebook part 2

#### Alternativer Code zu oben, aber mangels Rechenleistung nicht machbar: 
Im Test mit einem kleinen Datensatz hat der Code funkioniert:

In [52]:
# Alternative zu oben, nur afugrund der Rechenleistung nicht machbar

# df.insert(loc=16 , column='count_initiators', value=0)

In [53]:
#for index, row in df.iterrows(): 
#    df.loc[df.index, 'count_initiators'] = len(row.borrower_genders.split(','))

##### TEST mit kleinem Datensatz

In [54]:
#df_test = df.head(30)
#df_test

In [55]:
#df_test['neue_spalte'] = 0
#df_test

In [56]:
#for index, row in df_test.iterrows():
#    df_test.loc[index, row.neue_spalte] = len(row.borrower_genders.split(', '))

In [57]:
#df_test

In [58]:
#df_test.drop(columns='neue_spalte', inplace=True)
#df_test

In [59]:
#df_test.rename(columns={0: 'count_initiator'}, inplace=True)
#df_test

# Exportieren als CSV

In [ ]:
df.to_csv('crowdfunding_preprocess1.csv')